### DISCUSSION TITLE
Release check not working (apparently)

### DISCUSSION THREAD

**Discussion by EHRETic** [netbox-community/netbox#9276] [Q&A] [answered]
### NetBox version

v3.2.1

### Python version

3.9

### Steps to Reproduce

Hello there,

I've configured the following in my configuration file as mentioned [here](https://docs.netbox.dev/en/stable/configuration/optional-settings/#release_check_url) and I know there is a newer version available (3.2.2) but I don't see any sign in the main page (connected as admin, my only user)

```
# This repository is used to check whether there is a new release of NetBox available. Set to None to disable the
# version check or use the URL below to check for release in the official NetBox repository.
# RELEASE_CHECK_URL = None
RELEASE_CHECK_URL = 'https://api.github.com/repos/netbox-community/netbox/releases'
```

Python version is 3.9.6

Hope I didn't do anything wrong as [I already missed this option](https://github.com/netbox-community/netbox/issues/9237) in the documentation... 😉

### Expected Behavior

Should display a message to administrative users on the home page when a new release is detected.

### Observed Behavior

I don't see any error

**Comment by kkthxbye-code** [comment]
The check runs when the housekeeping command is executed. Make sure you have setup housekeeping.

https://docs.netbox.dev/en/stable/administration/housekeeping/

To verify you can trigger it manually:

`python manage.py housekeeping`

**Comment by EHRETic** [comment]
> `python manage.py housekeeping`

Hi,

Thanks a lot for helping. In that particular case, I'm getting an error with the command : 
```
Traceback (most recent call last):
  File "/opt/netbox-3.2.1/netbox/manage.py", line 8, in <module>
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'
```

I'm now wondering if there is not somewhere a misconfiguration from day 1 as I think I got the same missing django message by trying to install a map plugin (abandoned the idea for now). Netbox overall seems to work/update without any issue.

Netbox is installed on a dedicated VM, without docker and is installed that way (folder structure) on an AlmaLinux box and runs behind an NGINX reverse:
![image](https://user-images.githubusercontent.com/28634810/166219084-191a287a-0815-49d2-be01-97fc034100e0.png)

If somebody can help with this error too, that would be very nice, thanks in advance! 😉

**Comment by kkthxbye-code** [comment]
You need to activate the venv before running the command.

Please consult the documentation here:

https://docs.netbox.dev/en/stable/development/getting-started/#create-a-python-virtual-environment

**Reply by EHRETic** [reply, marked as answer]
> You need to activate the venv before running the command.
> 
> Please consult the documentation here:
> 
> https://docs.netbox.dev/en/stable/development/getting-started/#create-a-python-virtual-environment

Hi, 

Sorry but I really don't get this venv environnement thing and I can't find the manage.py script in this environnement.
And I really don't want to break my installation...

I finally created the cron command as mentioned in the docu... but would still understand the ground venv principles! 😉

**Comment by jeschebach** [comment]
Hi folks, I'm running Netbox v4.1.4.  I have the release check option working, but now we want to disable it and although the `RELEASE_CHECK_URL` environment variable has been removed from `/opt/netbox/netbox.env` and service restarted, the banner is still present in the UI. Wondering if we missed something, but haven't find further instructions in [the documentation](https://netboxlabs.com/docs/netbox/en/stable/configuration/miscellaneous/#release_check_url). 

Appreciate any help!

**Reply by jeschebach** [reply]
I've created this topic to talk about it https://github.com/netbox-community/netbox/discussions/19474
